src/components/SelectionList/BaseSelectionList.js

```diff
@@ -266,23 +266,20 @@ function BaseSelectionList({
         );
     };
 
-    /** Focuses the text input when the component mounts. If `props.shouldDelayFocus` is true, we wait for the animation to finish */
-    useEffect(() => {
-        if (shouldShowTextInput) {
-            if (shouldDelayFocus) {
+    /** Focuses the text input when the component comes into focus and after any navigation animations finish. */
+    useFocusEffect(
+        useCallback(() => {
+            if (shouldShowTextInput) {
                 focusTimeoutRef.current = setTimeout(() => textInputRef.current.focus(), CONST.ANIMATED_TRANSITION);
```

Comment:
Question:
There is `focusWithDelay` function in the codebase, maybe we can reuse it here instead of using setTimeout? 
I know that there we are using `ComposerFocusManager` which is not applicable here but maybe we can reuse similar logic here?

Reply:
As we discussed in DM - `runAfterInteractions` triggers immediately on web, making the animation feel janky

See: https://reactnative.dev/docs/interactionmanager